### PR TITLE
[SYCL][CUDA] tf32 matrix MAD impl using uint32_t

### PR DESCRIPTION
**Pull request by JackAKirk**
CUDA backend Implementation of tf32 MAD using the underlying 32 bit type, fully consistent with the existing matrix extension.

Integration test added here: https://github.com/intel/llvm-test-suite/pull/881

### CODE REVIEW

File: sycl/test/check_device_code/matrix/matrix-nvptx-fp19-test.cpp [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+float C[M * N];
+float D[M * N];
+
+int main() {
+
+  buffer<uint32_t, 1> bufA(A, range<1>(M * K));
+  buffer<uint32_t, 1> bufB(B, range<1>(K * N));
+  buffer<float, 1> bufC(C, range<1>(M * N));
+  buffer<float, 1> bufD(D, range<1>(M * N));
+
```
Comment (dkhaldi) [inline sycl/test/check_device_code/matrix/matrix-nvptx-fp19-test.cpp:46]: Can you add a complete example in test/matrix where you show the necessary "manual" conversion function from float to fp19(uint32) during initialization and then from fp19 to float during accumulation and verification?

File: sycl/test/check_device_code/matrix/matrix-nvptx-fp19-test.cpp [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+// M, N, K define the sizes of dimensions of the three matrix types (a, b,
+// accumulator) used per subgroup operation.
+constexpr int M = 16; // number of rows of accumulator,
+                      // number of cols of b.
+constexpr int N = 16; // number of cols of accumulator,
+                      // number of rows of a.
+constexpr int K = 8;  // number of cols of a/number of rows of b.
+
+uint32_t A[M * K];
```
Comment (dkhaldi) [inline sycl/test/check_device_code/matrix/matrix-nvptx-fp19-test.cpp:35]: add a comment that uint32 is used here as a storage for fp19

Comment (JackAKirk) [reply sycl/test/check_device_code/matrix/matrix-nvptx-fp19-test.cpp:35]: Yeah it's here: https://github.com/intel/llvm-test-suite/pull/881

File: sycl/test/check_device_code/matrix/matrix-nvptx-fp19-test.cpp (hunk 1 above)
Comment (JackAKirk) [reply sycl/test/check_device_code/matrix/matrix-nvptx-fp19-test.cpp:46]: Yeah it's here: https://github.com/intel/llvm-test-suite/pull/881
for the float to fp19 
```
uint32_t make_tf32(float const &x);
```
For the fp19 to float:

```
float tf32_to_fp32(uint32_t x);
```

(I'll rename both to e.g. make_fp19)

File: sycl/test/check_device_code/matrix/matrix-nvptx-fp19-test.cpp (hunk 2 above)
Comment (JackAKirk) [reply sycl/test/check_device_code/matrix/matrix-nvptx-fp19-test.cpp:36]: Thanks for the comments, I've updated both tests now.

**Comment by dkhaldi** [comment]
LGTM but we need to start adopting the name tf32 instead of fp19.
As soon as you make the change, I will approve.

**Review by dkhaldi — APPROVED** [review]

**Comment by JackAKirk** [comment]
This PR is no longer necessary: The complete tf32 implementation is now ready which can replace this PR: https://github.com/intel/llvm/pull/5870

**Pull request closed without merging**